src/tip-opengov.ts

```diff
@@ -72,6 +67,36 @@ export async function tipOpenGov(opts: {
           .catch(reject);
       });
   });
+
+  if (tipResult.success && polkassembly) {
+    void (async () => {
+      const condition = async (): Promise<boolean> => {
+        const lastReferendum = await polkassembly.getLastReferendumNumber(track.track.trackNo);
+        return lastReferendum !== undefined && lastReferendum >= referendumId.toNumber();
+      };
+      try {
+        bot.log.info(`Waiting until referendum ${referendumId.toString()} appears on Polkasssembly`);
+        await until(condition, 30_000);
+        polkassembly.logout();
+        await polkassembly.loginOrSignup();
+        await polkassembly.editPost(tipRequest.contributor.account.network, {
+          postId: referendumId.toNumber(),
```

Comment:
this referendumId still refers to the `await api.query.referenda.referendumCount();` which is prior to a new referendum, or I'm missing something?

Reply:
Correct, this is the `referendumCount()` prior to the new referendum - which is correct, because the query returns "the next free referendum index":

https://github.com/paritytech/substrate/blob/bc4055bb9fdd02ff4a97d33c6aaae3c635225b22/frame/referenda/src/lib.rs#L226-L228